From f79961e2be037dfc713fd95df8b23531e031c3f6 Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Sat, 5 Aug 2023 04:02:38 -0500
Subject: [PATCH 3/6] Fixup multiple defines

---
 lfd_encrypt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lfd_encrypt.c b/lfd_encrypt.c
index d73bcd7..5c6b209 100644
--- a/lfd_encrypt.c
+++ b/lfd_encrypt.c
@@ -64,8 +64,8 @@
 #define ENC_BUF_SIZE VTUN_FRAME_SIZE + 128 
 #define ENC_KEY_SIZE 16
 
-BF_KEY key;
-char * enc_buf;
+extern BF_KEY key;
+extern char * enc_buf;
 char * dec_buf;
 
 #define CIPHER_INIT		0
-- 
2.43.0

