From ecc99bc89cd7383b0014c37d5696459fc27a9a4a Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Sat, 5 Aug 2023 04:03:08 -0500
Subject: [PATCH 4/6] vtun client uses lot of CPU after a SIGHUP

---
 client.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/client.c b/client.c
index eb7a981..77ad928 100644
--- a/client.c
+++ b/client.c
@@ -133,6 +133,7 @@ void client(struct vtun_host *host)
 	   if (!vtun.quiet || errno != ETIMEDOUT)
 	      vtun_syslog(LOG_INFO,"Connect to %s failed. %s(%d)", vtun.svr_name,
 					strerror(errno), errno);
+	      client_term = 0;
         } else {
 	   if( auth_client(s, host) ){   
 	      vtun_syslog(LOG_INFO,"Session %s[%s] opened",host->host,vtun.svr_name);
@@ -145,6 +146,7 @@ void client(struct vtun_host *host)
 	      vtun_syslog(LOG_INFO,"Session %s[%s] closed",host->host,vtun.svr_name);
 	   } else {
 	      vtun_syslog(LOG_INFO,"Connection denied by %s",vtun.svr_name);
+	      client_term = 0;
 	   }
 	}
 	close(s);
-- 
2.43.0

