From 114ae56b549e306a960bdf9d55e974574e4bad51 Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Sat, 5 Aug 2023 04:03:38 -0500
Subject: [PATCH 5/6] retry limit

---
 tunnel.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tunnel.c b/tunnel.c
index 8428d53..008217d 100644
--- a/tunnel.c
+++ b/tunnel.c
@@ -127,6 +127,9 @@ int tunnel(struct vtun_host *host)
 	   opt=1;
 	   setsockopt(host->rmt_fd,IPPROTO_TCP,TCP_NODELAY,&opt,sizeof(opt) );
 
+	   opt=60000;
+	   setsockopt(host->rmt_fd,IPPROTO_TCP,TCP_USER_TIMEOUT,&opt,sizeof(opt) );
+
 	   proto_write = tcp_write;
 	   proto_read  = tcp_read;
 
-- 
2.43.0

