--- a/lfd_encrypt.h 1969-12-31 18:00:00.000000000 -0600
+++ b/lfd_encrypt.h 2025-02-18 21:38:24.182098423 -0600
@@ -0,0 +1,48 @@
+/*  
+    VTun - Virtual Tunnel over TCP/IP network.
+
+    Copyright (C) 1998-2008  Maxim Krasnyansky <max_mk@yahoo.com>
+
+    VTun has been derived from VPPP package by Maxim Krasnyansky. 
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+ */
+
+/*
+   Encryption module uses software developed by the OpenSSL Project
+   for use in the OpenSSL Toolkit. (http://www.openssl.org/)       
+   Copyright (c) 1998-2000 The OpenSSL Project.  All rights reserved.
+ */
+
+/*
+ * This lfd_encrypt module uses MD5 to create 128 bits encryption
+ * keys and BlowFish for actual data encryption.
+ * It is based on code written by Chris Todd<christ@insynq.com> with
+ * several improvements and modifications by me.
+ */
+
+/*
+ * The current lfd_encrypt module is based on code attributed above and 
+ * uses new code written by Dale Fountain <dpf-vtun@fountainbay.com> to 
+ * allow multiple ciphers, modes, and key sizes. Feb 2004.
+ */
+#ifndef _VTUN_LFD_ENCRYPT_H
+#define _VTUN_LFD_ENCRYPT_H
+
+int send_msg(int len, char *in, char **out);
+int recv_msg(int len, char *in, char **out);
+
+/* Send In-Band Message */
+int send_ib_mesg(int *len, char **in);
+/* Receive In-Band Message */
+int recv_ib_mesg(int *len, char **in);
+
+#endif /* _VTUN_LFD_ENCRYPT_H */
--- a/lfd_encrypt.c 2025-02-18 21:36:37.711551608 -0600
+++ b/lfd_encrypt.c 2025-02-18 21:41:41.029462075 -0600
@@ -44,8 +44,10 @@
 #include <strings.h>
 #include <string.h>
 #include <time.h>
+#include <netinet/in.h>
 
 #include "vtun.h"
+#include "lfd_encrypt.h"
 #include "linkfd.h"
 #include "lib.h"
