From 8210c8a05cc3df0336d80f6f4d1a38e0c15bd94b Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Tue, 18 Feb 2025 21:12:12 -0600
Subject: [PATCH 5/6] makefile: support destdir

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 7f8491e..282635d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,7 @@
 PREFIX = /usr/local
 MANDIR = $(PREFIX)/share/man
+DESTDIR =
+TARGET = $(DESTDIR)
 
 CDEBUGFLAGS = -Os -g -Wall
 
-- 
2.48.1

