From 5fcbbf23894666590556f67e6dedd766b8630a77 Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Sat, 22 Feb 2025 21:13:37 -0600
Subject: [PATCH 6/6] route: add log for install_route

---
 route.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/route.c b/route.c
index cd63676..0a747b1 100644
--- a/route.c
+++ b/route.c
@@ -488,7 +488,7 @@ install_route(struct babel_route *route)
         return;
     }
 
-    debugf("install_route(%s from %s)\n",
+    fprintf(stderr, "install_route(%s from %s)\n",
            format_prefix(route->src->prefix, route->src->plen),
            format_prefix(route->src->src_prefix, route->src->src_plen));
     rc = change_route(ROUTE_ADD, route, metric_to_kernel(route_metric(route)),
-- 
2.48.1

